### PR TITLE
Fix error on unbox deleted edits, add canned message tests

### PR DESCRIPTION
**Pull request by mlsteele**
The goal was to add more tests of canned messages. I stumbled across deleted edits failing to unbox. Turns out they couldn't be unboxed because of a constraint on unboxing that doesn't hold. Also those errors were hidden.

Changes:
- Don't check for the SupersededBy==Deleted constraint in `unbox`.
- Don't filter out errors when sending messages to the CLI / GUI. Done by adding an `includeAllErrors bool` parameter to `FilterByType`. Now those clients will show a row for every unbox error. This way if there's a bug in messages we will notice quickly, and if Eve's obtruding we'll get an indicator.
- Bump the MessageUnboxed cache version to nuke existing caches. People's existing caches contain unbox errors that would never otherwise heal.
- Add more canned message tests
- Fix some tests that had unbox errors but didn't notice (doesn't matter)

### CODE REVIEW

**Comment by mmaxim** [comment]
Fine with me, but I'm going to cc @patrickxb and @oconnor663, since they dealt more with the empty body case.

File: go/chat/boxer.go [hunk 1]
```diff
@@ -202,16 +202,13 @@ func (b *Boxer) unboxV1(ctx context.Context, boxed chat1.MessageBoxed, encryptio
...
+	// Whether the body is missing (deleted)
+	skipBodyVerification := (len(boxed.BodyCiphertext.E) == 0)
+
 	// decrypt body
 	// will remain empty if the body was deleted
 	var bodyVersioned chat1.BodyPlaintext
-	skipBodyVerification := false
-	if len(boxed.BodyCiphertext.E) == 0 {
-		if boxed.ServerHeader.SupersededBy == 0 {
-			return nil, NewPermanentUnboxingError(errors.New("empty body and not superseded in MessageBoxed"))
```
Comment (patrickxb) [inline go/chat/boxer.go]: why is this not an error?

**Comment by mmaxim** [comment]
Also, I'll just comment in here that I think the real problem might just be the process for deleting edits should be setting `supersededBy`, so that might be the preferred fix.

File: go/chat/boxer.go (hunk 1 above)
Comment (mlsteele) [inline go/chat/boxer.go]: We didn't actually keep up with this constraint.

If you do this
```
msg 1: TEXT
msg 2: EDIT msg 1
msg 3: DELETE msg 1
```

You get this
```
msg 1: TEXT deleted, supersededBy 3
msg 2: EDIT deleted, supersededBy 0 <-- oops
msg 3: DELETE, deletes [1,2]
```

This seems much easier than backfilling the database. We could potentially turn this check back on after doing that, for now I just wanted those valid edits to stop failing to unbox for the sake of these tests.

Comment (patrickxb) [inline go/chat/boxer.go]: I agree with your "oops", but doesn't msg 3 actually delete 1 and 2 from the database?

Comment (mlsteele) [inline go/chat/boxer.go]: Yep! So then a client downloads msg 2 and says "Wait a minute, this was deleted but not superseded! Fail to unbox."

Comment (patrickxb) [inline go/chat/boxer.go]: Ok, that makes sense.  Have we fixed it so supersededBy in the EDIT is not zero?

**Comment by patrickxb** [comment]
👍

File: go/chat/boxer.go (hunk 1 above)
Comment (mlsteele) [inline go/chat/boxer.go]: Not fixed yet, made a ticket CORE-4662

File: go/chat/boxer_test.go [hunk 2]
```diff
@@ -700,8 +700,9 @@ func TestChatMessageDeletedNotSuperseded(t *testing.T) {
 		// Delete the body
 		boxed.BodyCiphertext.E = []byte{}
 
-		_, err = boxer.unbox(context.TODO(), *boxed, key)
-		require.Error(t, err, "should not unbox with deleted but no supersededby")
+		unboxed, err := boxer.unbox(context.TODO(), *boxed, key)
+		require.NoError(t, err, "suprisingly, should be able to unbox with deleted but no supersededby")
```
Comment (oconnor663) [inline go/chat/boxer_test.go]: There's some context in this PR for why this is true? Maybe summarize it here in a comment?

File: go/chat/utils/utils.go [hunk 3]
```diff
@@ -344,19 +344,35 @@ func GetUnverifiedConv(ctx context.Context, g *libkb.GlobalContext, uid gregor1.
 	return inbox.ConvsUnverified[0], ratelim, nil
 }
 
-func FilterByType(msgs []chat1.MessageUnboxed, query *chat1.GetThreadQuery) (res []chat1.MessageUnboxed) {
-	if query != nil && len(query.MessageTypes) > 0 {
-		typmap := make(map[chat1.MessageType]bool)
+// FilterByType filters messages based on a query.
+// If includeAllErrors then MessageUnboxedError are all returned. Otherwise, they are filtered based on type.
+// Messages whose type cannot be determined are considered errors.
+func FilterByType(msgs []chat1.MessageUnboxed, query *chat1.GetThreadQuery, includeAllErrors bool) (res []chat1.MessageUnboxed) {
```
Comment (oconnor663) [inline go/chat/utils/utils.go]: Squicky gut reaction to boolean parameters, especially since this one has no existing callers for `false`. Your call.

**Comment by keybase-ci-visdiff** [comment]
The commits 4781abef38ee...625a666d7c49 introduce visual changes on linux. <details><summary>:mag_right: 9 changed</summary>

 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Folders_TLF-Normal_Public-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Folders_TLF-Normal_Public-625a666d7c49.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **Chat_Conversations_List-You_Rekey_Selected** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-You_Rekey_Selected-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-You_Rekey_Selected-625a666d7c49.png)  
   Chat_Conversations_List-You_Rekey_Selected comparison
 * Changed: **Chat_Conversations_List-You_Rekey** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-You_Rekey-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-You_Rekey-625a666d7c49.png)  
   Chat_Conversations_List-You_Rekey comparison
 * Changed: **Chat_Conversations_List-Selected_Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Selected_Normal-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Selected_Normal-625a666d7c49.png)  
   Chat_Conversations_List-Selected_Normal comparison
 * Changed: **Chat_Conversations_List-Selected_Muted** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Selected_Muted-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Selected_Muted-625a666d7c49.png)  
   Chat_Conversations_List-Selected_Muted comparison
 * Changed: **Chat_Conversations_List-Part_Rekey_Selected** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Part_Rekey_Selected-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Part_Rekey_Selected-625a666d7c49.png)  
   Chat_Conversations_List-Part_Rekey_Selected comparison
 * Changed: **Chat_Conversations_List-Part_Rekey** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Part_Rekey-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Part_Rekey-625a666d7c49.png)  
   Chat_Conversations_List-Part_Rekey comparison
 * Changed: **Chat_Conversations_List-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Normal-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Normal-625a666d7c49.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Normal.png)
 * Changed: **Chat_Conversations_List-Long_Top** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Long_Top-4781abef38ee.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Long_Top-625a666d7c49.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1488401758331-4781abef38ee-625a666d7c49-linux/Chat_Conversations_List-Long_Top.png)
</summary>

**Pull request merged**